go/vt/vttls/vttls.go

```diff
@@ -128,7 +128,7 @@ var onceByKeys = sync.Map{}
 
 // ClientConfig returns the TLS config to use for a client to
 // connect to a server with the provided parameters.
-func ClientConfig(mode SslMode, cert, key, ca, name string, minTLSVersion uint16) (*tls.Config, error) {
+func ClientConfig(mode SslMode, cert, key, ca, crl, name string, minTLSVersion uint16) (*tls.Config, error) {
```

Comment:
It's not really clear here from the signature, but the `ca` field here supports a list of PEM encoded CA certificates as a root store with multiple certificates. 

That might be confusing for the `crl` argument here if that only supports one CRL? What if `ca` contains 2 roots for validation (for example when changing the root CA, or with a larger system list), then it would not be possible to add multiple `crl`s for each CA? 

Or does this already work when multiple CRLs are PEM encoded in the given file?

Reply:
You are totally right. I did leave this undefined in the interest of time.

Currently, only the first PEM-encoded CRL is parsed from the file.

Happy to look at this, at the risk of delaying this PR since I have more priority work upcoming unfortunately

Reply:
> Currently, only the first PEM-encoded CRL is parsed from the file.

How hard would it be to read each PEM encoded CRL from the file so that that would provide a way to load multiple? This question is also specifically relevant combined with the other one around validating the signature of the CRL so it can be associated with the right CA then.

Reply:
I wouldn't say this is hard at all, just very intricate. It takes a lot of time to test and think through.

I have this commit as a proposal: https://github.com/Shopify/vitess/commit/cd042505330383d6410d4501baf9582db6a98569

Two notes that made me hesitant bringing it over here just yet:

1. I haven't had time to dig into what exactly we get as the `verifiedChains` argument. So I don't know if my assumption about `cert := chain[i], issuerCert := chain[i+1]` is entirely correct. It does achieve the validation of "whoever signed cert here also has signed the CRL".

2. Parsing multiple CRLs from a single file is gross.. I suppose it could use a contribution to `x509.ParseCRL(...)` method to parse multiple PEMs, similarly to `func (s *CertPool) AppendCertsFromPEM(pemCerts []byte)`

I'd love your thoughts/help 🙏

Reply:
Hey @dbussink, I did some more research and I'm happy enough to bring that commit in.

It seems like (1) is a safe assumption (for example, I see a team of security experts making a similar assumption in [this codebase](https://github.com/cossacklabs/acra/blob/a02704aff992b5b45b5b723960e58173f8483d87/network/crl.go#L527-L537)) and (2) can't easily be avoided.

Would you kindly review again?